src/api/cvc4cppkind.h

```diff
@@ -0,0 +1,2309 @@
+   */
+  JOIN_IMAGE,
+  /**
+   * Set identity.
+   * Parameters: 1
+   *   -[1]: Term of set Sort
+   * Create with:
+   *   mkTerm(Kind kind, Term child)
+   */
+  IDEN,
+
+  /* Strings --------------------------------------------------------------- */
+
+  /**
+   * String concat.
+   * Parameters: n > 1
+   *   -[1]..[n]: Terms of String sort
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  STRING_CONCAT,
+  /**
+   * String membership.
+   * Parameters: 2
+   *   -[1]..[2]: Terms of String sort
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  STRING_IN_REGEXP,
+  /**
+   * String length.
+   * Parameters: 1
+   *   -[1]: Term of String sort
+   * Create with:
+   *   mkTerm(Kind kind, Term child)
+   */
+  STRING_LENGTH,
+  /**
+   * String substr.
+   * Parameters: 3
+   *   -[1]: ???
+   *   -[2]: ???
+   *   -[3]: ???
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  STRING_SUBSTR,
+  /**
+   * String ???.
+   * Parameters: 2
+   *   -[1]: ???
+   *   -[2]: ???
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  STRING_CHARAT,
+  /**
+   * String contains.
+   * Parameters: 2
+   *   -[1]: ???
+   *   -[2]: ???
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  STRING_STRCTN,
+  /**
+   * String index-of.
+   * Parameters: 3
+   *   -[1]: ???
+   *   -[2]: ???
+   *   -[3]: ???
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  STRING_STRIDOF,
+  /**
+   * String replace.
+   * Parameters: 3
+   *   -[1]: ???
+   *   -[2]: ???
+   *   -[3]: ???
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  STRING_STRREPL,
+  /**
+   * String prefix-of.
+   * Parameters: 2
+   *   -[1]: ???
+   *   -[2]: ???
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  STRING_PREFIX,
+  /**
+   * String suffix-of.
+   * Parameters: 2
+   *   -[1]: ???
+   *   -[2]: ???
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  STRING_SUFFIX,
+  /**
+   * Integer to string.
+   * Parameters: 1
+   *   -[1]: Term of sort Integer
+   * Create with:
+   *   mkTerm(Kind kind, Term child)
+   */
+  STRING_ITOS,
+  /**
+   * String to integer (total function).
```

Comment:
If the string does not contain an integer or the integer is negative, the operator returns -1.